crates/bevy_ecs/src/observer/mod.rs

```diff
@@ -84,6 +84,17 @@ impl<'w, E, B: Bundle> Trigger<'w, E, B> {
     }
 }
 
+impl<'w, E: Debug, B: Bundle + Debug> Debug for Trigger<'w, E, B> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Trigger")
+            .field("event", &self.event)
+            .field("propagate", &self.propagate)
+            .field("trigger", &self.trigger)
+            .field("_marker", &self._marker)
```

Comment:
```suggestion
```
Debugging the PhantomData isn't useful

Reply:
I found `_marker: PhantomData<observers::Mine>,` part of the output useful, as `ComponentId(0)` doesn't give a lot of information.
But maybe debugging triggers of `Bundle`s without `Debug` derived is also useful?
Too bad there is still no impl specialization in rust.

Reply:
> But maybe debugging triggers of `Bundle`s without `Debug` derived is also useful?
> 
> Too bad there is still no impl specialization in rust.

The `Debug` bound on `B` doesn't really do anything because `PhantomData` doesn't require it for its own `Debug` impl. So it should be fine to remove it _and_ keep the `PhantomData`

Reply:
That's fine by me

Reply:
Removed the `Debug` bound, resolving this